### PR TITLE
Feat/hng 183 create badge component

### LINKED ISSUE #183: [FEAT] - Components → Badge
## Description
Create a Badge component with four variants (default, primary, success, error). The Badge component should accept a label value and an optional icon value.

---

## Acceptance Criteria
- Implement a Badge component with four variants: default, primary, success, and error.
- The component should display a label value.
- The component should optionally display an icon alongside the label.
- Ensure the Badge component is visually appealing and consistent across all variants.
- Maintain accessibility standards with appropriate aria attributes.

---

## Requirements
The Badge component should support the following props:
- `label: string` - Required label value to be displayed.
- `icon?: React.ReactNode` - Optional icon to be displayed alongside the label.
- `variant: 'default' | 'primary' | 'success' | 'error'` - Variant type to determine the badge style.

---

## Testing
- Verify that the Badge component renders correctly with different variants.
- Test that the component displays the label value properly.
- Ensure the optional icon is displayed correctly when provided.
- Conduct accessibility testing to ensure screen reader compatibility.

---

## Tasks
- [ ] **Component Development**: Develop the Badge component based on the specified requirements.
- [ ] **Props Implementation**: Implement support for `label`, `icon`, and `variant` props.
- [ ] **Styling**: Ensure consistent styling for all variants: default, primary, success, and error.
- [ ] **Unit Testing**: Write unit tests to validate component behavior.
- [ ] **Accessibility Testing**: Verify accessibility features are properly implemented.
- [ ] **Documentation**: Update documentation with usage examples and prop details.

---

#### Error + Icon
Size=medium, Color=Error, Icon=Dot

#### Success
Size=Large, Color=Success, Icon=False

#### Primary + icon
Size=Large, Color=Primary, Icon=Dot

#### default
Size=small, Color=Gray, Icon=False

---

## Link
[Figma Link](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2225-6357&t=XOIp4pqGzAyZNcxj-4)

### PR DESCRIPTION
**Pull request by vee2095**
# Description

This is a follow-up pull request from [#241](https://github.com/hngprojects/hng_boilerplate_nextjs/pull/241) and [#342](https://github.com/hngprojects/hng_boilerplate_nextjs/pull/342)

My pull requests have been reviewed three times, sometimes with conflicting directives, kindly refer to past reviews before requesting changes. Thank you.

**Closes #183**

# Changes proposed

## What were you told to do?
Create a Badge component with four variants (default, primary, success, error). The Badge component should accept a label value and an optional icon value

[FEAT] - Components → Badge

## What did you do?
Implement a Badge component with four variants: default, primary, success, and error.
Ensure the component displays a label value.
Ensure the component optionally displays an icon alongside the label.
Ensure the Badge component is visually appealing and consistent across all variants.
Maintain accessibility standards with appropriate aria attributes.

##  Checklist
- [x] Component Development: Develop the Badge component based on the specified requirements.
- [x] Props Implementation: Implement support for label, icon, and variant props.
- [x] Styling: Ensure consistent styling for all variants: default, primary, success, and error.
- [x] Unit Testing: Write unit tests to validate component behaviour.
- [x] Accessibility Testing: Verify accessibility features are properly implemented.
- [x] Documentation: Update documentation with usage examples and prop details.
- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

#### For usage information, kindly refer to the tsdocs in the [Badge component](https://github.com/OvyEvbodi/hng_boilerplate_nextjs/blob/feat/HNG-183-create-badge-component/src/components/common/Badge/Badge.tsx)

# Screenshots

Screenshot 2024-07-20 at 21 03 47

Screenshot 2024-07-20 at 21 03 11

Screenshot 2024-07-20 at 19 50 03

new tests
Screenshot 2024-07-21 at 15 24 56

Screenshot 2024-07-21 at 15 23 41

### CODE REVIEW

File: src/app/guides/page.tsx
Comment (skeby) [inline src/app/guides/page.tsx:1]: Remove this comment and move your test file to the test folder assap
Do this asap so we can merge

**Review by skeby — CHANGES_REQUESTED** [review]

**Comment by skeby** [comment]
@OvyEvbodi your lint test is failing. Please run `pnpm lint:fix` 
Run `pnpm lint` and attach the screenshot of the result

**Comment by vee2095** [comment]
Please refresh it on your end, I've attached a screenshot.

File: src/test/Badge.test.tsx
Comment (skeby) [inline src/test/Badge.test.tsx:1]: Ma
You're very funny oo
Why are your tests expect(true).toBeTruthy()?

Comment (vee2095) [reply src/test/Badge.test.tsx:1]: You can make your point without being rude. I've corrected them, I initially copied the boiler plate test I found in the repo, and was still learning React testing at the time.  You've even checked my tests before, and you didn't see the error. Thank you for pointing it out.

Comment (skeby) [reply src/test/Badge.test.tsx:1]: Sorry about that
Well approve your pr now
@oreoluwa212

**Review by skeby — APPROVED** [review]

**Pull request merged**